exp/controllers/azuremanagedmachinepool_reconciler.go

```diff
@@ -108,9 +139,9 @@ func (r *azureManagedMachinePoolReconciler) Reconcile(ctx context.Context, scope
 	}
 
 	var providerIDs = make([]string, len(instances))
-	for _, vm := range instances {
+	for i, vm := range instances {
 		vm := vm
```

Comment:
pretty sure I put this here, but is it actually necessary in this instance? though this mattered more for closures/goroutines

Reply:
might as well use `*instances[i].ID` since we already need `i`, then can drop vm from the loop. 

just style _shrugs_

Reply:
fixed